tests/testthat/test_predictions.R

```diff
@@ -0,0 +1,68 @@
+test_that("prediction with no categorical predictors", {
+  set.seed(1)
+  dat <- data.frame(x = runif(20), y = runif(20), z = runif(20))
+  gp_kern <- k_Exponential(D = 2)
+  gp_fit <- gpkm(z ~ x + y, data = dat, kernel = gp_kern)
+
+  preds <- gp_fit$pred(dat[, 1:2], se.fit = TRUE)
+
+  exp_ptype <-
+    structure(
+      list(mean = numeric(0), s2 = numeric(0), se = numeric(0)),
+      row.names = integer(0),
+      class = "data.frame")
+
+  expect_equal(preds[0,], exp_ptype)
+  expect_equal(nrow(preds), nrow(dat))
+
+})
+
+test_that("prediction with only categorical predictors", {
```

Comment:
I don't know if this is a feasible test case. Do you want folks to be able to fix purely categorical predictor models?